src/context/section-context.js

```diff
@@ -0,0 +1,31 @@
+import { createContext, useContext, useState, useEffect } from "react";
+import axios from "axios";
+
+const CategoryContext = createContext();
+
+const CatergoryContextProvider = ({ children }) => {
+  const [category, setCategory] = useState([]);
+
+  useEffect(() => {
+    const categoriesData = async () => {
+      try {
+        const response = await axios.get("/api/categories");
```

Comment:
You can put api calls in a seprate file so that you dont' make spelling mistake when writing